Sources/NIO/BaseSocketChannel.swift

```diff
@@ -493,19 +693,10 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         }
 
         // Fail all pending writes and so ensure all pending promises are notified
-        self._isOpen = false
         self.unsetCachedAddressesFromSocket()
         self.cancelWritesOnClose(error: error)
 
-        if !self.neverActivated {
-            becomeInactive0(promise: p)
-        } else if let p = p {
-            p.succeed(result: ())
-        }
-
-        if !self.neverRegistered {
-            pipeline.fireChannelUnregistered0()
-        }
+        self.lifecycleManager.moveState(event: .close, promise: p)()
```

Comment:
Do we want to wrap these in convenience methods that let the states and inputs become private to `lifecycleManager`? e.g. `self.lifecycleManager.close()`.